### PR TITLE
Implemented FITS header reader and added quotation marks in CMakeLists

### PR DESCRIPTION
**Pull request by aalexr**
Check this GUI features before merge:

- [x] Positive and negative `BITPIX` images
- [x] Zoom, rotate and flip
- [x] Level control
- [x] Color maps
- [x] Open new image in the same window and in new window

### CODE REVIEW

**Comment by matwey** [comment]
Could you please add your `outlook.com` email to your GitHub account settings? Now GitHub cannot link the email with your account.

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -272,6 +276,26 @@ void MainWindow::openFileHere() {
 	}
 }
 
+void MainWindow::viewHeader() {
+	auto& header_unit = state_->fits().header_unit();
+	header_view_ = new QDialog(this, Qt::WindowCloseButtonHint);
```
[the remark this answers is not in the document]
Comment (aalexr) [reply src/mainwindow.cpp]: Thanks for the review!

When method is called multiple times, pointer to the previous dialog is overriden by new one. I agree that it is not the best solution, but fortunately there is no memory leak because of `Qt::WA_DeleteOnClose`, I suppose. I've check this using Visual Studio's memory profiler.

If we declare `header_view_` as local variable there are two possible behaviors. The first one - we call `header_view_.exec()` that blocks parent window until child window is closed. The second - we call  `header_view_.show()` which does not block, but we lose control under local variable-pointer and get memory leak. Probably, `Qt::WA_DeleteOnClose` helps, but, as you wrote below, you don't like that. Anyway, we need to look for another solution.

Oh, I didn't really think about changing file while headers are opened. That's a silly mistake)) In my defence, I can say that mayble a pro user might want to compare different headers without keeping ALL fits files opened.

[the remark this answers is not in the document]
Comment (matwey) [reply src/mainwindow.cpp]: I am fine with the behavior which I see in d99cdac5ef005cb09804350c155da0d1f49b6696
I think the feature could be implemented in that way for now. It is better than nothing and eventually may be refactored.

File: include/fits.h [hunk 2]
```diff
@@ -100,6 +100,10 @@ class FITS {
 				qvariant_cast<T>(it->second) : def);
 		}
 
+		std::map<QString, QString>::const_iterator cbegin() const { return headers_.cbegin(); }
+		std::map<QString, QString>::const_iterator cend() const { return headers_.cend(); }
+		size_t size() const { return headers_.size(); }
+
```
[the remark this answers is not in the document]
Comment (aalexr) [reply include/fits.h]: I can't understand why we need `begin()` `end()`. I declared `cbegin(), cend()` on purpose to emphasise that iterator is readonly! Can you explain why I should declare this instead? As a regular user I expected read-write behaviour when I call 'begin()` `end()`

[the remark this answers is not in the document]
Comment (matwey) [reply include/fits.h]: Mutability is a property of variable, not functions. Member functions in C++ can be overloaded by `const` specification, refer to `std::vector<>` for example:

```
iterator begin() { return ...; }
iterator end() { return ...; }
const_iterator begin() const { return ...; }
const_iterator end() const { return ...; }
```

```
void fun(const std::vector<>& x) {
x.begin(); // return type is const_iterator
}
```

```
void fun(std::vector<>& x) {
x.begin(); // return type is iterator
}
```

`cbegin()`/`cend()` have been introduced **only** in C++11 as synonyms for `begin() const`. So,  I think it would be a good style to provide pre-C++11 conventional functions.

Also, for the best of my knowledge `std::begin(x)` newer asks for `x.cbegin()` (I may be wrong here), which is important for range-based for loops.
Have you tried that the following is compiled:

```
for (auto x: headers) {

}
```

?

[the remark this answers is not in the document]
Comment (aalexr) [reply include/fits.h]: Finally, I got it. Range-based loop is not working without `begin()`.

File: src/mainwindow.cpp [hunk 3]
```diff
@@ -264,7 +264,7 @@ void MainWindow::setState(std::unique_ptr<MainWindowState>&& new_state) {
 }
 
 void MainWindow::openFileHere() {
-	const auto filename = QFileDialog::getOpenFileName(this, tr("Open FITS file in current window"));
+	const auto filename = QFileDialog::getOpenFileName(this, tr("Open FITS file in current window"), tr(""), tr("*.fits;*.fit"));
```
Comment (matwey) [inline src/mainwindow.cpp]: Could you please put this into separate commit?

Comment (hombit) [reply src/mainwindow.cpp]: Please, provide an option to show and select all files, some FITS files may have different extension.

File: src/mainwindow.cpp [hunk 4]
```diff
@@ -272,6 +276,31 @@ void MainWindow::openFileHere() {
 	}
 }
 
+void MainWindow::viewHeaders() {
+	auto& header_unit = state_->fits().header_unit();
```
Comment (matwey) [inline src/mainwindow.cpp]: Could you please use
```
const auto& header_unit = ...
```
instead. AFAIU, we are not going to modify header_unit in this routine
